webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/MainActivity.kt

```diff
@@ -246,21 +263,34 @@ class MainActivity : AppCompatActivity() {
     private inner class YdsWebClient : WebViewClient() {
         // Detect the URL that indicates that YDS flow is finished
         // and close the app
-        override fun shouldOverrideUrlLoading(
-                view: WebView?, request: WebResourceRequest?
-        ): Boolean {
-            if (request?.url?.toString() == "https://www.yoti.com/") {
+        override fun onPageCommitVisible(view: WebView?, url: String?) {
+            super.onPageCommitVisible(view, url)
+            if (url == FINISH_SESSION_URL) {
                 AlertDialog.Builder(this@MainActivity)
                         .setTitle("YDS Session")
-                        .setMessage("Session finished OK")
+                        .setMessage("Session finished")
                         .setPositiveButton("OK") { _, _ -> this@MainActivity.finish() }
                         .show()
-
-                return true
             }
-            request?.url?.toString()?.let { Log.d(TAG, it) }
-
-            return super.shouldOverrideUrlLoading(view, request)
         }
     }
+
+    private fun showOptionsDialog() {
+        if (sessionBottomSheet != null) return
```

Comment:
should we consider checking `sharedialog.isShowing` before return?

Reply:
We're setting it to null when it is dismissed, so I think that verification is not needed :)